bftengine/src/preprocessor/RequestProcessingState.cpp

```diff
@@ -115,17 +115,24 @@ void RequestProcessingState::detectNonDeterministicPreProcessing(const uint8_t *
 }
 
 void RequestProcessingState::handlePreProcessReplyMsg(const PreProcessReplyMsgSharedPtr &preProcessReplyMsg) {
+  SCOPED_MDC_CID(cid_);
   const auto &senderId = preProcessReplyMsg->senderId();
   if (preProcessReplyMsg->status() == STATUS_GOOD) {
-    numOfReceivedReplies_++;
     const auto &newHashArray = convertToArray(preProcessReplyMsg->resultsHash());
+    // Do not add a duplicate signature from the same sender to the list.
+    // insert returns a pair, where the second is a boolean denoting whether the insertion took place.
+    // TODO - identify the scenarios that can cause duplicate replies to arrive.
+    if (!(seen_senders_[newHashArray]).insert(preProcessReplyMsg->senderId()).second) {
+      LOG_INFO(logger(), "Signature was already accepted for this sender " << KVLOG(senderId, reqSeqNum_, clientId_));
```

Comment:
should log level be INFO or WARNING?

Reply:
I've validated with Guy, this scenario is normal